backend/src/index.ts

```diff
@@ -76,17 +83,27 @@ async function main() {
       // We know that block number will not exceed 53-bit size integer
       let lastProcessedBlock = Number(totals.count.toBigInt()) - 1;
 
+      const relayParams = {
+        logger,
+        target,
+        httpApi,
+        batchBytesLimit: BATCH_BYTES_LIMIT,
+        batchCountLimit: BATCH_COUNT_LIMIT,
+      };
+
+      let relay;
+
       if (chainConfig.downloadedArchivePath) {
+        const db = levelup(rocksdb(`${chainConfig.downloadedArchivePath}/db`), { readOnly: true });
```

Comment:
it seemed to me that `ChainArchive` should not know anything about db implementation, because we might want to change it for example. Also, if we use real db that would be an integration test, I was trying to start with unit tests, because they are just simpler.

Reply:
In that case we need a separate DB abstraction, I really dislike that we are still bound to RocksDb API, yet pretend we are not. And in that sense `ChainArchive` was that quite simple abstraction over RocksDb, it does very little, but provides domain-specific typed API.